subsys/bluetooth/mesh/transport.c

```diff
@@ -727,7 +727,7 @@ static int sdu_recv(struct bt_mesh_net_rx *rx, uint8_t hdr, uint8_t aszmic,
 			.aszmic = aszmic,
 			.src = rx->ctx.addr,
 			.dst = rx->ctx.recv_dst,
-			.seq_num = rx->seq,
+			.seq_num = seg->seq_auth & 0xffffff,
```

Comment:
This is not right. `seg` will only be defined when this function is called from `trans_seg`. What's the motivation for moving the net_rx->seq set here?

Reply:
The sequence will be modified, which will cause the sequence number of the segment message stored in the friend queue to be lower than that of the previously stored segment message, which will cause the LPN node to start a replay attack after receiving the message

Reply:
Okay, I see now. When we change the `net_rx->seq` in `trans_seg()`, it will be wrong when we enqueue it in the friend queue in the end of `bt_mesh_trans_recv()`. 

The change here doesn't work for single segment messages though, as they pass `seg=NULL` to the `sdu_recv()` function.
Suggest changing it to this, which maps nicely to the rule in the spec:

```suggestion
			.seq_num = seg ? (seg->seq_auth & 0xffffff) : rx->seq,
```